### ISSUE TITLE
Adding the same exception to the xray trace handler in different subsegments throws a null reference exception

### ISSUE THREAD

**Issue by jon-armen** [aws/aws-xray-sdk-dotnet#209]
This code should work, however currently throws a null reference exception because the existing cause does not have an exception associated.

```
            try
            {
                recorder.BeginSubsegment("child1");
                try
                {
                    try
                    {
                        recorder.BeginSubsegment("child2");
                        throw new AmazonServiceException();
                    }
                    catch (AmazonServiceException e)
                    {
                        recorder.AddException(e);
                        recorder.EndSubsegment();
                        throw;
                    }
                }
                catch (AmazonServiceException e)
                {
                    recorder.AddException(e);
                    recorder.EndSubsegment();
                    throw new EntityNotAvailableException("Dummy message", e);
                }
            }
            catch (EntityNotAvailableException e)
            {
                recorder.AddException(e);
                recorder.EndSegment();
            }
```

**Comment by lupengamzn** [comment]
Hi @jon-armen ,

Thank you for bringing this up

This is indeed an edge case issue in X-Ray SDK while serializing duplicate exception. In the second `catch` statement, as the `AmazonServiceException` from previous step has already been recorded, SDK will only record the [Cause](https://github.com/aws/aws-xray-sdk-dotnet/blob/d2b64125ecdd734af220795d5b102bd96b037981/sdk/src/Core/Strategies/DefaultExceptionSerializationStrategy.cs#L156) to the exception descriptor. When the `AmazonServiceException` is wrapped as an inner exception and passed to next catch statement, the SDK will try to figure out if this inner exception has been recorded or not by checking the [Exception](https://github.com/aws/aws-xray-sdk-dotnet/blob/d2b64125ecdd734af220795d5b102bd96b037981/sdk/src/Core/Strategies/DefaultExceptionSerializationStrategy.cs#L192) from the descriptor, which, as I mentioned before, will not be recorded as only the `Cause` will be recorded for duplicate exception. This is why you see the null reference exception.

We'll work on this and get it fixed soon.

**Comment by jon-armen** [comment]
Awesome, thank you @lupengamzn .  If it helps, I submitted #210 as a potential fix.

**Comment by srprash** [comment]
Released in v2.11.0
